src/server/game/Entities/GameObject/GameObjectData.h

```diff
@@ -487,7 +487,7 @@ struct GameObjectTemplate
             case GAMEOBJECT_TYPE_GOOBER:     return goober.noDamageImmune != 0;
             case GAMEOBJECT_TYPE_FLAGSTAND:  return flagstand.noDamageImmune != 0;
             case GAMEOBJECT_TYPE_FLAGDROP:   return flagdrop.noDamageImmune != 0;
-            default: return true;
+            default: return false;
```

Comment:
what does this change ?

Reply:
Just the default value, CannotBeUsedUnderImmunity is not used anywhere, it could be renamed and changed after.

Reply:
Careful! For `GAMEOBJECT_TYPE_CHEST` this function should always return `true` **unless** "DamageImmuneOK" property is set, which was only created in later expansions:
https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Entities/GameObject/GameObjectData.h#L112
And thus for 3.3.5 chests can be considered to always return `true`.

Reply:
Both DamageImmuneOK and noDamageImmune  are not implemented I just renamed the bool.

Reply:
No, you also inverted the (unimplemented) result of the method for all types apart from the ones used in switch. Whether or not it wasn't used before, for chests it was returning the correct value, and now it isn't, so it would be great if you fixed this unintended side-effect. Implementing it is just a matter of time, it isn't difficult, but if you let this mistake remain the code, it might confuse future implementers.

Reply:
Reverted

Reply:
Why revert it instead of just adding `case GAMEOBJECT_TYPE_CHEST: return true;`...

Reply:
It must be fine with the last change

Reply:
Wonderful, thank you!